pkg/sharing/shared.go

```diff
@@ -175,4 +175,31 @@ func GetSharedDocsBySharingID(inst *instance.Instance, sharingID string) ([]couc
 	return result, nil
 }
 
+// GetSharingsByDocType returns the sharingids related to the
+// given docType
+func GetSharingsByDocType(inst *instance.Instance, docType string) ([]string, error) {
+	var req = &couchdb.AllDocsRequest{
+		StartKey: docType + "/",
+		EndKey:   docType + "/\uFFFF",
+	}
+
+	var docs []*SharedRef
+	err := couchdb.GetAllDocs(inst, consts.Shared, req, &docs)
+	if err != nil {
+		return nil, err
+	}
+	// Use a map to easily avoid duplicates
+	keys := make(map[string]bool)
+	var sharingIDs []string
+	for _, doc := range docs {
+		for id := range doc.Infos {
```

Comment:
You don't look at the `removed` flag of Infos, so the files/documents that have been removed/deleted/taken out of the sharing will be included. Is it what you want to do? The description of the endpoint is not really clear on that.

I'd have not done like that, but both points of view can be defended.

Reply:
From my perspective, even a deleted document could have the legitimity to be included as it has been shared at some point. 
That being said, this could be confusing for the querier, so we might filter out the removed documents here and add an option `include_removed=true` in the future, if needed.